src/app/static/src/tests/components/modelOutput/modelOutput.test.ts

```diff
@@ -304,8 +304,124 @@ describe("ModelOutput component", () => {
         expect(store.state.plottingSelections.colourScales.output).toBe(bubbleSizeScales);
     });
 
+    it("commits updated selections from barchart and orders them according to filter", () => {
```

Comment:
This test is a little hard to reason about. I think I see what you're doing, but it's a little counterintuitive, because you're setting up the expected selection order to the comit as a re-order of the order emitted from the barchart? I think it could be made more readable by just renaming 'updatedBarchartSelections' to 'expectedBarchartSelections' and either just including the expected options objects literally or else setting them as a copy of the order in `testBarchartFilters`  (which is closer to the logic of what's happening)

Similarly for the test of nested filters.